### PR TITLE
Issue19 call backend apis

### PR DESCRIPTION
**Pull request by k4yluh**
Implementing controller functions by calling backend APIs for P0 issues
#19

### CODE REVIEW

File: navigation/screens/facility-profile.tsx
Comment (dylantompkins) [inline navigation/screens/facility-profile.tsx:1]: why did you commit commented out code?

File: navigation/screens/home.tsx
Comment (dylantompkins) [inline navigation/screens/home.tsx:1]: why did you create a `home.tsx`? we already have `main.tsx`. what does this have to do with calling backend apis?

**Review by dylantompkins — CHANGES_REQUESTED** [review]

File: navigation/screens/facility-profile.tsx
Comment (k4yluh) [reply navigation/screens/facility-profile.tsx:1]: to pass lint

File: .vscode/c_cpp_properties.json
Comment (dylantompkins) [inline .vscode/c_cpp_properties.json:1]: delete all the .vscode files. they are your 357 c++ config and not relevant to this project

File: controllers/userController.ts [hunk 1]
```diff
@@ -1,9 +1,9 @@
 import axios from "axios";
-import { User as UserProfile } from "@/types/user";
-import { Facility as BathroomProfile } from "@/types/facility";
-import { Review } from "@/types/review";
+import { User as UserProfile } from "../types/user";
+import { Facility as BathroomProfile } from "../types/facility";
 
 const port = process.env.PORT || 8081;
```
Comment (dylantompkins) [inline controllers/userController.ts:5]: this should probably be 3000 like the other ones. maybe move the environment variable declations into their own file so that you don't have to change them in 3 places? duplicate code makes it easy to forget changing things

File: controllers/userController.ts [hunk 2]
```diff
@@ -21,11 +21,12 @@ export async function registerUser(
   username: string,
   email: string,
   password: string,
-  profile: typeof Image,
+  profile: string,
```
Comment (dylantompkins) [inline controllers/userController.ts:24]: this type should match the comment above. there's 3 possible types here that you have to choose from.

1. typeof Image
2. string
3. Optional<Image>

Since the controller's purpose is to model the backend endpoints, the types should represent as closely as possible what the backend need. Think about that and take a look at the backend to decide.

File: controllers/userController.ts [hunk 3]
```diff
@@ -98,78 +99,20 @@ export async function getUserProfile(username: string): Promise<UserProfile> {
  */
 export async function createBathroom(
   username: string,
-  opHours: string,
+  openHr: Date,
+  closeHr: Date,
   description: string,
   tags: string[],
-  images: string[],
+  image: string,
```
Comment (dylantompkins) [inline controllers/userController.ts:106]: see above comment on this type

File: package.json
Comment (dylantompkins) [inline package.json:1]: you definitely need axios as a dependency. why did you remove it? what are those other packages?

**Review by dylantompkins — CHANGES_REQUESTED** [review]
Good job with adjusting the URLs and adding parameters!

Have you tested this against the backend? Ideally I'd like to see some integration tests but manual testing would be fine for now

File: controllers/userController.ts [hunk 4]
```diff
@@ -12,7 +12,7 @@ const host = process.env.HOST || "localhost";
  * @param {string} username - The username of the user
  * @param {string} email - The email address of the user
  * @param {string} password - The password of the user *encrypted
- * @param {Optional<Image>} profile - Profile display image
+ * @param {string} profile - Profile display image
```
Comment (dylantompkins) [inline controllers/userController.ts:15]: Kayla and I discussed in class that we'll use `string` for now since that's what the backend says. Kayla pointed out that it probably should be optional throughout though, since there is a default pfp.

Kayla, please create a follow up task to change frontend and backend to make this Optional

File: controllers/userController.ts [hunk 5]
```diff
@@ -94,7 +94,7 @@ export async function getUserProfile(username: string): Promise<UserProfile> {
  * @param {string} opHours - The hours of operation of the bathroom (days and times)
  * @param {Optional<string>} description - Description of the bathroom
  * @param {string[]} tags - List of tags to filter the bathroom by
- * @param {Optional<Image>} images - Uploaded images of the bathroom
+ * @param {string} image - Uploaded images of the bathroom
```
Comment (dylantompkins) [inline controllers/userController.ts:97]: Similar to comment on profile picture, Kayla and I decided to match the backend for now. Kayla thinks that this would be better as a list of strings though, as there could be 0..many images associated. I agree.

Kayla, please create a follow up task for this as well.

File: package-lock.json
Comment (dylantompkins) [inline package-lock.json:1]: remove

File: navigation/screens/home.tsx
Comment (dylantompkins) [reply navigation/screens/home.tsx:1]: this file showed back up again. not sure what it is, it seems to overlap a lot with main.tsx

Comment (k4yluh) [reply navigation/screens/home.tsx:1]: pretty sure it came back from pulling from main because I did not add it back...

Comment (k4yluh) [reply navigation/screens/home.tsx:1]: actually just checked and its not in main but I definitely deleted it when you told me to...

**Comment by k4yluh** [comment]
I am implementing unittests to test the controllers using Jest and Axios-mock-adapter, but having trouble passing lint tests because lint does not recognize jest functions...

**Comment by dylantompkins** [comment]
```
diff --git a/eslint.config.js b/eslint.config.js
index 18bcc31..0e9b504 100644
--- a/eslint.config.js
+++ b/eslint.config.js
@@ -11,7 +11,7 @@ export default [
   eslint.configs.recommended,
   ...tseslint.configs.recommended,
   {
-    files: ["components/__tests__/**"],
+    files: ["components/__tests__/**", "controllers/__tests__/**"],
     ...jest.configs["flat/recommended"],
   },
 ];
```

@k4yluh 
the above change solves most of your linting errors, though there are some legimate linting errors still

File: components/__tests__/__snapshots__/StyledText-test.js.snap
Comment (dylantompkins) [inline components/__tests__/__snapshots__/StyledText-test.js.snap:1]: what is this?

File: jest.config.js
Comment (dylantompkins) [inline jest.config.js:1]: where did you get this config from?

**Review by dylantompkins — CHANGES_REQUESTED** [review]
Tests look great!! Just a couple files which I'm not sure what they do. If you can answer them that would be great.

The last thing I want to see is this actually tested against the backend running on the same computer. Here's how I would achieve that:
1. Choose a button that is easy to get to
2. replace whatever it was doing with a call to the contoller
3. put the result of the call to the controller in an Alert, so that it's easy for us to see
4. try with all the different controller functions to make sure that they work
5. don't commit the code with those temporary changes to the button

File: jest.config.js
Comment (k4yluh) [reply jest.config.js:1]: why whats wrong with it? I couldn't add the jest to the package.json so I had to make a separate config file

File: components/__tests__/__snapshots__/StyledText-test.js.snap
Comment (k4yluh) [reply components/__tests__/__snapshots__/StyledText-test.js.snap:1]: think it got added accidentally when I was installing

File: jest.config.js
Comment (dylantompkins) [reply jest.config.js:1]: nothing wrong, was just wondering how it generated it

**Comment by k4yluh** [comment]
all controller functions are able to be utilized -- tested against backend 

One point though -- for the create bathroom (post) API, it was the only one that needed me to call using my frontend port instead of my backend port, which I could see was running on 3000 (but the API required me to run on 8081 (my frontend port)), but other POST APIs were able to be called on backend port

File: .env.vault
Comment (dylantompkins) [inline .env.vault:1]: did you create an env vault for the frontend? if not, this shouldn't be here

File: components/__tests__/StyledText-test.js
Comment (dylantompkins) [inline components/__tests__/StyledText-test.js:1]: why did you delete this?

**Review by dylantompkins — CHANGES_REQUESTED** [review]
looking pretty good. a couple of weird things that I commented on. also want to dive deeper into the port issue you said in your most recent comment

**Review by dylantompkins — APPROVED** [review]
